squad-create-skipfile-reproducers

```diff
@@ -0,0 +1,237 @@
+    if branch_name in linux_stable_rc_branches:
+        return f"{linux_stable_rc_branches[branch_name]}-{branch_name}"
+    elif branch_name in linux_mainline_next_branches:
+        return f"{branch_name}-{linux_mainline_next_branches[branch_name]}"
+    else:
+        raise Exception(
+            "Unknown branch name {0}. \nSupported branch names: \n\t- {1}".format(
+                branch_name, "\n\t- ".join(supported_branch_names())
+            )
+        )
+
+
+def parse_args(raw_args):
+    parser = argparse.ArgumentParser(
+        description="Produce TuxRun or TuxPlan reproducers for the LTP skipfile."
+    )
+
+    parser.add_argument(
+        "--group",
+        required=True,
+        help="The name of the SQUAD group.",
+    )
+
+    parser.add_argument(
+        "--allow-unfinished",
+        required=False,
+        action="store_true",
+        default=False,
+        help="Allow fetching of reproducers where the build is marked as unfinished.",
+    )
+
+    parser.add_argument(
+        "--branches",
+        required=False,
+        default=supported_branch_names(),
+        nargs="+",
+        help="A list of SQUAD branches to be tested.",
+    )
+
+    parser.add_argument(
+        "--build-names",
+        required=False,
+        default=["gcc-12-lkftconfig", "gcc-\d\d-lkftconfig"],
+        nargs="+",
+        help="The list of accepted build names (for example, gcc-12-lkftconfig). Regex is supported.",
+    )
+
+    parser.add_argument(
+        "--debug",
+        required=False,
+        action="store_true",
+        default=False,
+        help="Display debug messages.",
+    )
+
+    parser.add_argument(
+        "--device-names",
+        required=False,
+        default=["qemu-armv7", "qemu-arm64", "qemu-i386", "qemu-x86_64"],
+        nargs="+",
+        help="The list of device names (for example, qemu-arm64).",
+    )
+
+    parser.add_argument(
+        "--search-build-count",
+        required=False,
+        default=10,
+        type=int,
+        help="The number of builds to fetch when searching for a reproducer.",
+    )
+
+    parser.add_argument(
+        "--skipfile-url",
+        required=False,
+        default="https://raw.githubusercontent.com/Linaro/test-definitions/master/automated/linux/ltp/skipfile-lkft.yaml",
+        help="URL of the skipfile to test.",
+    )
+
+    parser.add_argument(
+        "--suite-name",
+        required=False,
+        default="ltp-syscalls",
+        help="The suite name to grab a reproducer for.",
+    )
+
+    return parser.parse_args(raw_args)
+
+
+def run(raw_args=None):
+    start = time.time()
+    args = parse_args(raw_args)
+
+    skipfile = squadutilslib.get_file(args.skipfile_url)
+
+    reason_list = []
+    reproducer_scripts = []
+
+    with open(skipfile) as f:
+        reasons = yaml.load(f, Loader=yaml.FullLoader)
+    for reason in reasons["skiplist"]:
+        if reason["boards"] == "all" or "all" in reason["boards"]:
```

Comment:
if no boards or a specific board what do we do then?

Reply:
Added check for cases where the list in the skipfile entry and our boards list have common members